pyblnet/blnet.py

```diff
@@ -106,6 +106,51 @@ def _turn(self, digital_id, value, can_node=None):
         else:
             raise EnvironmentError('Can\'t set values with blnet web disabled')
 
+    def get_value(self,type='digital', ret='value', name=None, id=None, cached=None):
+        """
+          higher level interface to get a value or mode by name or id
+          ret: can be 'value', 'mode'
+          returns the value  if ret='value' or the mode if ret='mode' as first return value
+                  and the dictionary containing name, value, id and mode as a second return value
+          cached: in order to prevent polling data from BLNet with every call, 
+                  the data can be fetched once and stored and passed to this function
+
+
+        """
+        val=None
+        dic=None
+        if name is None and id is None: return val
+        if cached==None:
+          cached=self.fetch()
+        for key,v in cached[type].items():
+           if v['name']==str(name) or name==None: 
+             if v['id']==str(id) or id==None:
```

Comment:
Please replace this with an `and` combination

Reply:
I'm not sure whether the lefthand arguments are always strings (especially 'id'). Did you check this?

Reply:
to be safe, I add a str() around the left side.

Reply:
Sorry, I do not know how to change this to an and combination. Maybe, I misunderstand